app/furniture/marketplace/order_policy.rb

```diff
@@ -2,12 +2,20 @@
 
 class Marketplace
   class OrderPolicy < Policy
-    class Scope < ApplicationScope
+    class Scope
+      attr_accessor :scope, :shopper
+      delegate :person, to: :shopper
+
+      def initialize(shopper, scope)
```

Comment:
The first argument -- `shopper` in this case -- is always just `current_user`, controlled by Pundit right? (Getting more familiar with pundit.)

Reply:
Well, it's a little more nuanced. 

We `includePundit::Authorization` in our `ApplicationController` this grants us the `Pundit::Authorization#policy_for` and  `Pundit::Authorization#policy_scope` methods.

Both of these methods call `#pundit_user` when providing `Policy` and `Scope` objects with their `user`. We define `#pundit_user` in our `ApplicationController`as an alias `#current_person`; which returns either a `Person` or a `Guest`, depending on if the person on the other end of the browser has gone through the Sign In flow.

So what I've done is said "OK `OrderPolicy::Scope`... Screw `#pundit_user`, `Person`, and `Guest`! Take a `Shopper` instead!" and instantiated the `OrderPolicy::Scope` directly rather through the `policy_scope` helper method.

In my first try, I had attempted to override `#pundit_user` in the `Marketplace::OrdersController`; which failed miserably  because it started passing `Shopper`s everywhere, including the `RoomPolicy` and `RoomPolicy::Scope` that gets invoked in `ApplicationController#current_room`...

TL/DR: While there is no `current_user` in Convene; there is a `pundit_user`, which `Pundit::Authorization#policy_for` and `Pundit::Authorization::policy_scope` use; which we define in `ApplicationController` as the `current_person

Reply:
I guess this works for this specific bugfix, but I think it is a bad idea to break the expected Pundit interfaces like this. This will fall apart as soon as someone tries to do a standard `policy_scope(Order).where(...)` somewhere, and this will need to be reworked.